Source/Core/Core/CoreTiming.cpp

```diff
@@ -317,10 +319,19 @@ void CoreTimingManager::Advance()
     std::pop_heap(m_event_queue.begin(), m_event_queue.end(), std::greater<Event>());
     m_event_queue.pop_back();
 
+    g_watchdog.SetMarker("Throttle");
     Throttle(evt.time);
+
+    if (evt.type != nullptr && evt.type->name != nullptr)
+      g_watchdog.SetMarker(*evt.type->name);
```

Comment:
As `Marker` is a `std::string_view`, how sure are you that the store within `CoreTimingManager::m_event_types` won't be deleted before your reference is deleted?

I'm not sure if a persistent storage of string_view is a good idea or just asking for trouble.
However using the string itself (instead of the pointers) is too large overhead especially for profiler.
Another option is to use persistent objects within the storage of whatever shall be profiled and a pointer to them.